evennia/contrib/actions/handlers.py

```diff
@@ -0,0 +1,752 @@
+                "your turn to begin with.")
+        else:
+            self.owner.msg("You cannot finish your turn without being in " +
+                "turn-based mode to begin with.")
+
+    def stop(self, ongoing=True, queued=False):
+        """
+        Cancels the character's ongoing actions if ongoing is set to True,
+        and removes the character's queued actions if queued is set to True.
+        """
+        if queued:
+            # First stop all enqueued actions if the action calls for it, 
+            # to ensure that cancelling the ongoing actions will not load
+            # any actions from the queue
+            for action in self.owner.actions.list:
+                self.owner.actions.remove(action)
+
+        if ongoing:
+            # Then cancel all ongoing actions
+            for action in self.owner.location.actions.list:
+                if action['owner'] == self.owner:
+                    action['at_cancel'](action)
+
+
+    def unpuppet(self):
+        """
+        Cancels all of the character's queued and ongoing actions, then
+        removes the character's alert, which causes the room to attempt to
+        enter real-time mode.
+        """
+        self.stop(ongoing=True, queued=True)
+        self.turnbased = False
+
+
+class RoomActionHandler(object):
+    """
+    Lists the ongoing action in the room, keeps track of turn-related
+    information and provides methods for 
+    """
+
+    def __init__(self, owner):
+        """
+        sets owner (Room), the room to which this handler is attached
+        """
+        self.owner = owner
+
+
+    def setup(self, override=False):
+        if not self.owner.db.actions or override:
+            self.owner.db.actions = {}
+
+        if not self.owner.db.actions.has_key("list") or override:
+            self.list = []
+
+        if not self.owner.db.actions.has_key("time") or override:
+            self.time = time() 
+
+        if not self.owner.db.actions.has_key("mode") or override:
+            self.mode = "RT"
+
+        if not self.owner.db.actions.has_key("turnof") or override:
+            self.turnof = None
+ 
+        if not self.owner.db.actions.has_key("acted") or override:
+            self.acted = []
+
+        if not self.owner.db.actions.has_key("view") or override:
+            self.view = None
+
+        if override:
+            for script in self.owner.scripts.all():
+                if type(script) == ActionSystemScript:
+                    self.owner.scripts.stop(script)
+        if not self.owner.scripts.get("ActionSystemScript"):
+            self.owner.scripts.add(ActionSystemScript, autostart=False)
+            self.script = self.owner.scripts.get("ActionSystemScript")[0]
+            self.script.start()
+
+    @property
+    def list(self):
+        """
+        the list of all ongoing actions in the room
+        """
+        return self.owner.db.actions['list']
+
```

Comment:
Btw, according to PEP8, only single blank space between methods in a class (double line breaks between classes).